.clang-format

```
@@ -0,0 +1,41 @@
+Language: Cpp
+AccessModifierOffset: 0
+AlignConsecutiveMacros: true
+BreakBeforeBraces: Linux
+IndentWidth: 2
+ConstructorInitializerIndentWidth: 2
+AlignAfterOpenBracket: Align
+Standard: c++11
+IndentAccessModifiers: true
+IndentCaseBlocks: true
+IndentCaseLabels: true
+IndentGotoLabels: true
+IndentExternBlock: Indent
+UseTab: Never
+ReflowComments: true
+QualifierAlignment: Left
+ReferenceAlignment: Pointer
+PointerAlignment: Left
+InsertBraces: true
+NamespaceIndentation: All
+LineEnding: LF
+IntegerLiteralSeparator:
+    Binary: 4
+    BinaryMinDigits: 8
+    Decimal: 3
+    DecimalMinDigits: 4
+    Hex: 2
+    HexMinDigits: 6
+AlwaysBreakTemplateDeclarations: true
```

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#alwaysbreaktemplatedeclarations - This should just be `BreakTemplateDeclarations`

```suggestion
BreakTemplateDeclarations: Yes
```

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#breaktemplatedeclarations - I think we should add something about Templates to `CONTRIBUTING.md`, although I don't think we rely on templates in many places in Re::Solve

---

[i deliberately reverted it to the deprecated form](https://github.com/ORNL/ReSolve/pull/151/commits/3f61b211a1b376171d740a026e359287a16cfacc) because my local clang-format version is older than 19

---

Sounds like we should specify what `clang-format` version we support / expect

---

a solid minimum would be 16. some of the other features i'm using in the `.clang-format` here were added in 16 and that's the version i'm using locally